server/schedule/filter/filters.go

```diff
@@ -77,6 +77,9 @@ func Target(opt opt.Options, store *core.StoreInfo, filters []Filter) bool {
 	storeAddress := store.GetAddress()
 	storeID := fmt.Sprintf("%d", store.GetID())
 	for _, filter := range filters {
+		if filter == nil {
```

Comment:
When will `filter` be `nil`? I did not find this.

Reply:
see `NewPlacementLeaderSafeguard`, it would return nil if placementRules is not enabled.